subworkflows/local/parse_input.nf

```diff
@@ -0,0 +1,104 @@
+/*
+ * Check input samplesheet or folder and get read channels
+ */
+
+params.options = [:]
+
+include { parse_samplesheet } from '../../modules/local/parse_samplesheet' addParams( options: params.options )
+
+workflow PARSE_INPUT {
+    take:
+    input // file.tsv or folder
+	single_end
+    multipleSequencingRuns
+    extension
+    
+    main:
+	if ( input.toString().toLowerCase().endsWith("fasta") ) {
+		// Fasta input directely for classification
+		ch_fasta = Channel.fromPath(input, checkIfExists: true)
+		ch_reads = Channel.empty()
+	} else {
+		ch_fasta = Channel.empty()
+		
+		if ( input.toString().toLowerCase().endsWith("tsv") ) {
```

Comment:
Are you sure that `TSV` is still OK? I understood all pipelines should use `CSV` as samplesheets now

Reply:
Where did you get the info that `CSV` is preferred (I seem to have missed that)? Personally I prefer tab-separation, but do not care too much (because I even more prefer the folder input). We could probably allow (additionally?) a comma separated input with `csv` ending...

Reply:
in the nf-core slack channel in January: https://nfcore.slack.com/archives/C01BB6T15NF/p1610447810196000

Reply:
Thanks, this sounds to me not mandatory but advisable to change to CSV. Will discuss this.